linkerd/protocol/http/src/test/scala/io/buoyant/linkerd/protocol/http/ZipkinTracePropagatorTest.scala

```diff
@@ -0,0 +1,305 @@
+    req.headerMap.add("b3", "a3ce929d0e0e4736-00f067aa0ba902b7-1")
+
+    val trace = ztp.traceId(req)
+    //b3 has been removed
+    assert(req.headerMap.get("b3").isEmpty)
+    // check "x-b3-traceid" and "x-b3-spanid" and "x-b3-sampled" have been added to request
+    assert(req.headerMap.keys == Set("x-b3-traceid", "x-b3-spanid", "x-b3-sampled"))
+
+    trace match {
+      case Some(tid) => {
+        assert(tid.traceId.toString().equals("a3ce929d0e0e4736"))
+        assert(tid.spanId.toString().equals("00f067aa0ba902b7"))
+        assert(tid.sampled.toString().equals("Some(true)"))
+      }
+      case None => {
+        assert(false)
+      } // "traceId does not exist"
+    }
+
+    // after "x-b3-" have been added check they have the same values as above
+    val trace2 = ztp.traceId(req)
+    assert(trace == trace2)
+
+    //process sample
+  }
+
+  test("get traceid from a b3 single header - four fields - child span on debug") {
+    //b3: a3ce929d0e0e4736-00f067aa0ba902b7-d-5b4185666d50f68b
+
+    val ztp = new ZipkinTracePropagator()
+    val req = Request()
+    req.headerMap.add("b3", "a3ce929d0e0e4736-00f067aa0ba902b7-d-5b4185666d50f68b")
+
+    val trace = ztp.traceId(req)
+    //b3 has been removed
+    assert(req.headerMap.get("b3").isEmpty)
+    // check "x-b3-traceid" and "x-b3-spanid" and "x-b3-flags" and "x-b3-parentspanid" have been added to request
+    assert(req.headerMap.keys == Set("x-b3-traceid", "x-b3-spanid", "x-b3-flags", "x-b3-parentspanid"))
+
+    trace match {
+      case Some(tid) => {
+        assert(tid.traceId.toString().equals("a3ce929d0e0e4736"))
+        assert(tid.spanId.toString().equals("00f067aa0ba902b7"))
+        assert(tid.parentId.toString().equals("5b4185666d50f68b"))
+        assert(tid.flags.toLong == 1)
+      }
+      case None => {
+        assert(false)
+      } // "traceId does not exist"
+    }
+
+    //test x-b3- headers have been added so we can get the same trace from them
+    val trace2 = ztp.traceId(req)
+    assert(trace == trace2)
+  }
+
+  test("get traceid from a b3 single header - four fields - not sampled child span") {
+    //b3: a3ce929d0e0e4736-00f067aa0ba902b7-d-5b4185666d50f68b
+
+    val ztp = new ZipkinTracePropagator()
+    val req = Request()
+    req.headerMap.add("b3", "a3ce929d0e0e4736-00f067aa0ba902b7-0-5b4185666d50f68b")
+
+    val trace = ztp.traceId(req)
+    //b3 has been removed
+    assert(req.headerMap.get("b3").isEmpty)
+    // check "x-b3-traceid" and "x-b3-spanid" and "x-b3-sampled" and "x-b3-parentspanid" have been added to request
+    assert(req.headerMap.keys == Set("x-b3-traceid", "x-b3-spanid", "x-b3-sampled", "x-b3-parentspanid"))
+
+    trace match {
+      case Some(tid) => {
+        assert(tid.traceId.toString().equals("a3ce929d0e0e4736"))
+        assert(tid.spanId.toString().equals("00f067aa0ba902b7"))
+        assert(tid.parentId.toString().equals("5b4185666d50f68b"))
+        assert(tid.sampled.toString() == "Some(false)")
```

Comment:
Use .contains here. This is an option that should contain a bool that is false so `assert(tid.sampled.contains(false))` should do.

Reply:
Done, used sampled.contains(false), much cleaner.